components/raftstore/src/store/peer.rs

```diff
@@ -821,6 +821,16 @@ where
         }
         let log_term = self.get_index_term(log_idx);
 
+        let last_log_term = entries.last().unwrap().term;
+        if last_log_term > self.term() {
+            // Hack: In normal flow, when leader sends the entries, it will use a term that's not less
+            // than the last log term. And follower will update its states correctly. For merge, we append
+            // the log without raft, so we have to take care of term explicitly to get correct metadata.
+            self.raft_group
+                .raft
+                .become_follower(last_log_term, INVALID_ID);
```

Comment:
Better add some logs to indicate that the peer becomes a follower because of appending merge entries.